components/VercelChat/PureAttachmentsButton.tsx

```diff
@@ -0,0 +1,123 @@
+import { PaperclipIcon } from "lucide-react";
+import { Button } from "../ui/button";
+import { useVercelChatContext } from "@/providers/VercelChatProvider";
+import { Attachment } from "@ai-sdk/ui-utils";
+import { useRef } from "react";
+
+function PureAttachmentsButton() {
+    const { setAttachments } = useVercelChatContext();
+    const fileInputRef = useRef<HTMLInputElement>(null);
+    const MAX_FILES = 10;
+
+    const uploadFile = async (file: File) => {
+        // Only allow image files for now
+        const allowedTypes = ['image/jpeg', 'image/png', 'image/gif', 'image/webp'];
+        if (!allowedTypes.includes(file.type)) {
+            console.error('File type not supported:', file.type);
+            return;
+        }
```

Comment:
if you're filtering this in the `input` why do you need this logic here for `allowedTypes`?
- can this code be removed? if no, why not?

Reply:
This is a double check in JS code in case the user tries to bypass the allowed files in the file input through the browser web inspector.

The user can easily bypass it by removing the accept attribute from the file input. If we do not have a check in JS, then this input will also accepts video files, almost anything, and upload tothe  file storage database

Reply:
that sounds like an extreme edge case yeah? in that case, what would happen? why not let the hacket try to send the custom fily type?